pkg/testpackage/testpackage.go

```diff
@@ -14,14 +14,24 @@ const (
 	SkipRegexpFlagDefault = `(export|internal)_test\.go`
 )
 
+const (
+	AllowPackagesFlagName    = "allow-packages"
+	AllowPackagesFlagUsage   = `comma separated list of packages that don't end with _test that tests are allowed to be in`
+	AllowPackagesFlagDefault = `main`
+)
+
 // NewAnalyzer returns Analyzer that makes you use a separate _test package
 func NewAnalyzer() *analysis.Analyzer {
 	var (
-		skipFileRegexp = SkipRegexpFlagDefault
-		fs             flag.FlagSet
+		skipFileRegexp   = SkipRegexpFlagDefault
+		allowPackagesStr = AllowPackagesFlagDefault
+		fs                 flag.FlagSet
 	)
 
 	fs.StringVar(&skipFileRegexp, SkipRegexpFlagName, skipFileRegexp, SkipRegexpFlagUsage)
+	fs.StringVar(&allowPackagesStr, AllowPackagesFlagName, allowPackagesStr, AllowPackagesFlagUsage)
+
+	allowedPackages := strings.Split(allowPackagesStr, ",")
```

Comment:
Test fails because this line should be moved bolow, inside `Run` func.

Flag package changes value of `allowPackagesStr` only after `NewAnalyzer` returns. Before that `allowPackagesStr` equals to `main`. So `allowedPackages` is always equal to `main`.